### ISSUE TITLE
Don't warn about a function not invoked if the function is exported

### ISSUE THREAD

**Issue by ale5000-git** [koalaman/shellcheck#3035]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2329
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

dummy_1=''
export dummy_1

dummy_2()
{
  :
}
export -f -- dummy

exit 0
```

#### Here's what shellcheck currently says:
[SC2329](https://www.shellcheck.net/wiki/SC2329) (info): This function is never invoked. Check usage (or ignored if invoked indirectly).

#### Here's what I wanted or expected to see:
Like variables, it would be nice if it doesn't warn also about function it they are exported.
One real life example may be this:
```sh
#!/usr/bin/env bash

_export_func_supported='false'
(
  dummy()
  {
    :
  }
  export -f -- dummy 2> /dev/null
) && _export_func_supported='true' || true

exit 0
```
